gse-app/src/main/java/com/powsybl/gse/app/ProjectPane.java

```diff
@@ -585,17 +599,33 @@ private MenuItem createDeleteProjectNodeItem(List<? extends TreeItem<Object>> se
         return menuItem;
     }
 
-    private MenuItem createRenameProjectNodeItem(TreeItem selectedTreeItem) {
+    private MenuItem createRenameProjectNodeItem(TreeItem<Object> selectedTreeItem) {
         MenuItem menuItem = new MenuItem(RESOURCE_BUNDLE.getString("Rename"), Glyph.createAwesomeFont('\uf120').size("1.1em"));
         menuItem.setOnAction(event -> {
             Optional<String> result = RenamePane.showAndWaitDialog((ProjectNode) selectedTreeItem.getValue());
             result.ifPresent(newName -> {
                 if (selectedTreeItem.getValue() instanceof ProjectNode) {
                     ProjectNode selectedTreeNode = (ProjectNode) selectedTreeItem.getValue();
                     selectedTreeNode.rename(newName);
-                    refresh(selectedTreeItem.getParent());
-                    treeView.getSelectionModel().clearSelection();
-                    treeView.getSelectionModel().select(selectedTreeItem);
+                    TreeItem<Object> parentItem = selectedTreeItem.getParent();
+                    refresh(parentItem);
+                    parentItem.getChildren().addListener(new ListChangeListener<TreeItem<Object>>() {
+                        @Override
+                        public void onChanged(Change<? extends TreeItem<Object>> change) {
+                            if (change.getList().size() != 1 || change.getList().get(0).getValue() != null) {
+                                treeView.getSelectionModel().clearSelection();
+                                treeView.getSelectionModel()
+                                        .select(change.getList()
+                                                .stream()
+                                                .filter(item -> item.getValue() instanceof ProjectNode &&
+                                                        ((ProjectNode) item.getValue()).getId().equals(selectedTreeNode.getId()))
+                                                .findFirst()
+                                                .orElseThrow(NoSuchElementException::new));
+                                refreshTabs();
+                                change.getList().removeListener(this);
+                            }
+                        }
+                    });
```

Comment:
This is a hack: you provide a listener that removes itself to keep the selected node. I think there is a better solution for this: change the value of the selected node, and refresh it should be sufficient

Reply:
The selected node is detroyed by the refresh, so there is no way until refresh (which is ansynchronous) complete, to known what will be the new TreeItem instance.

Reply:
Also there is also the same issue when we move an element. The tab view title contains le full path of the element and  is not refreshed

Reply:
And also even when renaming it does not cover all cases.
Suppose you have 2 nested folders test1 and test2 and a case inside test2.
open the case, rename test1 and the tab title is not refreshed.